src/server/clientiface.cpp

```diff
@@ -396,8 +381,6 @@ void RemoteClient::GetNextBlocks (
 	// emerging, continue next time browsing from here
 	if (nearest_emerged_d != -1) {
 		new_nearest_unsent_d = nearest_emerged_d;
-	} else if (nearest_emergefull_d != -1) {
```

Comment:
This was not needed. Either nearest_emerged_d is set, then it is used and nearest_emergefull_d, or it is not set, in which case nearest_emergefull_d is the nearest (if set). The distinctions was pointless and made the code harder to understand.